lib/instrumentation/span.js

```diff
@@ -1,5 +1,6 @@
 'use strict'
 
+const { executionAsyncId } = require('async_hooks')
```

Comment:
Question: Are there any perf. concerns with the `executionAsyncId()` function calls?  With two calls per span, a span heavy service/application is going to be calling the function a moderate amount.  I'm not familiar enough the async_hooks implementation to know how heavy the call to `executionAsyncId` is.

Reply:
```js
const Benchmark = require('benchmark')
const { executionAsyncId } = require('async_hooks')

console.log(`# Benchmark executionAsyncId()`);

var suite = new Benchmark.Suite({
  onStart: function () {
  }
})

suite
  .add('executionAsyncId', function() {
    executionAsyncId()
  })

  .on('cycle', function(event) {
    console.log(String(event.target));
  })
  .on('complete', function() {
    console.log('Fastest is: ' + this.filter('fastest').map('name'));
  })
  .run({ async: false });
```

```
$ node bench-executionAsyncId.js
# Benchmark executionAsyncId()
executionAsyncId x 930,504,788 ops/sec ±1.42% (85 runs sampled)
```

Comparing to other benchmarks, that is the same speed order as just doing a field lookup in a mapping. Checking the source, that appears to be exactly what it is: https://github.com/nodejs/node/blob/v16.9.1/lib/internal/async_hooks.js#L561-L563

Note that we call `executionAsyncId()` currently for every `agent.currentTransaction` or `agent.currentSpan` (when asyncHooks=true).

Reply:
👍